src/components/flyout/deviceTagFlyout.js

```diff
@@ -215,10 +215,7 @@ class DeviceTagFlyout extends React.Component {
     const { devices } = this.props;
     const { newTags, deletedTagNames } = this.state;
     const deviceIds = devices.map(({ Id }) => `'${Id}'`).join(',');
-    const tags = {
-      ...this.state.commonTagValues,
-      ...this.state.overiddenDeviceTagValues
-    };
+    const tags = { ...this.state.commonTagValues };
```

Comment:
I think this code could be made a bit cleaner. Change the `deleteExistingTag` method to store the results in an object where the key is the tag and the value is null. If you do that, lines 218 through 228 become the following (note adding commonTagValues to line 216).

```
const { newTags, deletedTagNames, commonTagValues } = this.state;
...
const tags = { 
   ...commonTagValues,
   ...deletedTagNames,
   ...newTags
};
```